### PR TITLE
638 canvas 404

### PR DESCRIPTION
**Pull request by tvay11**
Not right at the moment. I am going to need some help with the exist function in Folio.java.

### CODE REVIEW

**Comment by thehabes** [comment]
When you know you still need to work on it, make it a Draft Pull Request.  I am changing it to that since I know that's the case for this PR.  Then I will test it.

**Comment by thehabes** [comment]
@tvay11 I tested without looking at the code, here is the error I got.

```com.mysql.jdbc.JDBC4PreparedStatement@5be81a3d: select * from folios where pageNumber=** NOT SPECIFIED **
 edu.slu.tpen.servlet.CanvasServlet.doGet null
        java.sql.SQLException: No value specified for parameter 1
                at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1072)
                at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:986)
                at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:981)
                at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:926)
                at com.mysql.jdbc.PreparedStatement.checkAllParametersSet(PreparedStatement.java:2546)
                at com.mysql.jdbc.PreparedStatement.fillSendPacket(PreparedStatement.java:2522)
                at com.mysql.jdbc.PreparedStatement.fillSendPacket(PreparedStatement.java:2448)
                at com.mysql.jdbc.PreparedStatement.executeQuery(PreparedStatement.java:2202)
                at org.apache.commons.dbcp.DelegatingPreparedStatement.executeQuery(DelegatingPreparedStatement.java:93)
                at textdisplay.Folio.exists(Folio.java:357)
```

It is telling you exactly what is wrong and exactly on what line of code.  If you tested this locally, you should have supplied the error in the description of your Draft Pull Request.  If you didn't test...testing your own code is step 1.  You have now waited hours for help over an obvious error.

I expect you can handle this.  You need to specify a value for `pageNumber` in your query.  Look at the [PreparedStatement Java API](https://docs.oracle.com/javase/7/docs/api/java/sql/PreparedStatement.html) to figure out how to specify a value for that query parameter.  

If you get passed this but run into another error, let us know by posting the error from your tests.  If it isn't working but you aren't getting an error, then give us more detail on what is wrong.

**Comment by tvay11** [comment]
Show 404 on '/canvas/', negative numbers, and a non-existent folio.
show 200 on existing Folio.
Screen Shot 2022-10-31 at 2.02.53 PM
Screen Shot 2022-10-31 at 2.03.08 PM
Screen Shot 2022-10-31
Screen Shot 2022-10-31 at 2 03 08 PM
Screen Shot 2022-10-31 at 2 04 24 PM
Screen Shot 2022-10-31 at 2 03 00 PM
Screen Shot 2022-10-31 at 2 03 03 PM
Screen Shot 2022-10-31 at 2 03 05 PM
 at 2.04.24 PM
Screen Shot 2022-10-31 at 2.03.00 PM
Screen Shot 2022-10-31 at 2 02 53 PM

Screen Shot 2022-10-31 at 2.03.03 PM
Screen Shot 2022-10-31 at 2.03.05 PM

File: src/java/edu/slu/tpen/servlet/CanvasServlet.java [hunk 1]
```diff
@@ -51,30 +52,35 @@ public class CanvasServlet extends HttpServlet{
...
-                    getLogger(CanvasServlet.class.getName()).log(SEVERE, null, "No ID provided for canvas");
-                    resp.sendError(SC_NOT_FOUND);
-                }
-            } catch (NumberFormatException | SQLException | IOException ex) {
-                getLogger(CanvasServlet.class.getName()).log(SEVERE, null, ex);
-                throw new ServletException(ex);
-            }
-
+         else {
+              System.out.println("!!!!");
```
Comment (thehabes) [inline src/java/edu/slu/tpen/servlet/CanvasServlet.java:71]: Don't want this in production!

File: src/java/textdisplay/Folio.java [hunk 2]
```diff
@@ -345,6 +345,33 @@ public static int createFolioRecord(String collection, String pageName, String i
...
+    public static boolean exists(int folioID) throws SQLException{
+       boolean result = false;
+       String query = "select * from folios where pageNumber=?";
+       Connection j = null;
+       PreparedStatement ps = null;
+      try {
+         j = getConnection();
+         ps = j.prepareStatement(query);
+         ps.setInt(1,folioID);
+         //ps.setString(1, canvas);
```
Comment (thehabes) [inline src/java/textdisplay/Folio.java:357]: Don't want these comments in production!

File: src/java/textdisplay/Folio.java [hunk 3]
```diff
@@ -345,6 +345,33 @@ public static int createFolioRecord(String collection, String pageName, String i
...
+         //
+         //
+         ResultSet rs = ps.executeQuery();
+         if (rs.next()) {
+            int ans = rs.getInt(11);
+            if(ans>0){
+                return true;
+            }
+
```
Comment (thehabes) [inline src/java/textdisplay/Folio.java:366]: Cleanup this whitespace, and any others you notice.

--Note that you can use a [linter](https://lightrun.com/java/top-10-java-linters/) before submitting code, which will 'prettify' it and take care of this kind of stuff automatically.  You should be able to find one you can plug into Netbeans.

**Review by thehabes — APPROVED** [review]
Tested and fully functional.  Somehow you also took care of the weird 500 happening when we handed in id's that were too long, like http://localhost:8080/TPEN/canvas/111111111111111111111111111111111 - that was also 404!  Nice.

The changes requested are just cleanup, so this is still approved.  However, make sure to get rid of commented out stuff and extra System.out()s and whitespaces in the future.

**Review by Woodlnsoul — APPROVED** [review]

**Pull request merged**